woocommerce/woocommerce-gutenberg-products-block
# Products block: tax_query should be calculated only if __woocommerceAttributes is valid

I've added a check to calculate `$tax_query` only if a valid `__woocommerceAttributes` attribute is present, which resolves issue #8765.

```
$tax_query        = $is_valid_attributes ? $this->merge_tax_queries( $attributes_query, $visibility_query ) : [];
```

Additionally, I've refactored the update_rest_query method in the ProductQuery.php file to enhance code readability. This involves simplifying conditional expressions with variables and using ternary operators to streamline logic.

Fixes #8765 





### Testing
1. Create a new post.
2. Add a Products block to the post.
3. Save and publish the post.
4. In the Editor, open the browser console and run the following code:
```
const fetchProducts = () => {
    const paths = [
        "/wp/v2/product?per_page=100",
        "/wp/v2/product?per_page=100&product_cat=20",
    ];

    Promise.all(paths.map(async path => {
        return new Promise((resolve, reject) => {
            wp.apiFetch({path: path})
            .then(products => {
                resolve({
                    query: path,
                    products: products.map(product => {
                        return {name: product.title.rendered, category: product.product_cat.join(',')};
                    })
                });
            });
        });
    })).then(output => {
        console.log(output);
    });
}
fetchProducts();
```
5. After executing the code above, you will receive an array. Verify that:
     1. The first item in the array contains all the products.
     2. The second item in the array includes only items with the specified category ID, which is `20` in the code above. If you don't have any products with `product_cat=20`, feel free to change the category ID. 
6. Switch to the `trunk` branch and repeat steps 4 and 5. Due to the bug, you will observe that both array items contain all products, meaning that `product_cat=20` won't work for the second query path ("/wp/v2/product?per_page=100&product_cat=20").


### WooCommerce Visibility




### Changelog

> Fix: tax_query should be calculated only if __woocommerceAttributes is valid
